### ISSUE TITLE
Controllers drifting / rotating on their own and all devices nudging forward/backward

### ISSUE THREAD

**Issue by Wully616** [NOLOVR/NOLO-Driver-For-Windows#11]
When using the NoloVR i've found in some instances when I hold the controller still, one of them will slowly rotate orientation. Double pressing the power button fixes it.

Other times when moving the controllers around in a space they will "jump", they don't lose tracking, but their positioning is totally incorrect.

Finally, when standing still, facing the base station, I notice both the controllers and my position from the headset marker pulsing forward and back. At first I thought it was the base station vibrating (it does vibrate a lot!) but I secured it down and it hasn't resolved the issue.

**Comment by POEDaley** [comment]
I can confirm that I've experienced this same thing with the controllers jumping back and forth.  Just holding the controllers they will sometimes move nearly 1m away for a couple seconds, then back again.  My headset marker does the same thing.

**Comment by Hettylool** [comment]
@Wully616   @POEDaley 
Hi,
for this question,you can take a video to me.

**Comment by POEDaley** [comment]
I certainly can.  I'll have something for you within the next 24 hours.

**Comment by Hettylool** [comment]
@POEDaley @Wully616 
 pleace before that 
you can check the power of the device first,
and check the light of device if controller and headset maker 's light are all long green light.

**Comment by POEDaley** [comment]
I made sure I gave the controller a full charging, and all lights were green.  Here is the video I took after taking the controllers and base station off of the charger and testing in SteamVR:

https://youtu.be/WLQE33gB38Q

**Comment by Wully616** [comment]
@POEDaley Exact same problems i'm seeing. It subtly does it with my headmarker too, I couldn't really tell from your video if you had that issue.

**Comment by POEDaley** [comment]
Yeah, it happens to the headset marker as well.  It did happen a little bit there, but it was mostly just the one controller.  At my friends house, it was pretty severe.  It may be environmental, but I've now heard it happening in many locations so perhaps some noise filter or something needs to be applied.

**Comment by out51der** [comment]
Hi I can confirm I had a bad drift possibly by controllers in Google Earth , and couple of other titles. In Google Earth globe could not stop rotating like a button was stuck or something , but in other steam apps would be fine and then drift away suddenly 2-3 meters away? Any suggestions?
Thanks

**Comment by mnbvcxzl90** [comment]
Ay it's Daily Tech. Channel made me want to buy a nolo in the first place :). Anyway, I can confirm I'm also having a serious problem with controllers randomly teleporting away and back. Having quite a bit of tracking lag with the controllers as well, and very noticeable latency with head tracking if I set it to use nolo for position&rotation.

**Comment by nanospork** [comment]
I'm also getting an issue where my controllers sometimes rotate weirdly. It's not even always yaw rotation. Playing right now, my right-hand controller has rolled about 45 degrees clockwise. It does not reset when double-tapping the power button.

**Comment by Hettylool** [comment]
@nanospork 
everytime nolo restart,the rotation of controller will change at once,double click will reset it,but it will reset in VR GAME,not in the real world,so if you paly steamvr make sure the steamvr front is real world's front

**Comment by nanospork** [comment]
@Hettylool I'm well aware of that. The controllers aren't yawing though - at least not very much. When they do yaw, the double click reset works.

The problem is that sometimes they _roll_, and the double-tap reset doesn't seem to work in that situation.

**Comment by nanospork** [comment]
@Hettylool The _roll_ might be the way I had the controllers set up in Unity. I will get back to you on that. The double-tap reset button does work great though!

**Comment by rod78** [comment]
Same issue here. It's pretty drastic for me in all games and apps. I have to reset the controllers at least 8 times in one game of longbow.

**Comment by NexonSU** [comment]
Same issue. Can you fix it already? It's really annoying!
https://www.youtube.com/watch?v=SB1Jnkmfnsw

**Comment by LyuzhouZ** [comment]
For controllers orientation, please make sure the controller orientation calibration has been completed according to the following steps - 

Step 1. Stand facing base station, the distance between base station and the player is about 2m with no block in the middle, please have the turned-on controllers in hand and swirl around the wrist a bit first (this is to collect the surrounding magnetic field);

Step 2. With headset marker and controllers facing base station, double-click power button on each of the controllers respectively, to calibrate the controller orientation;

**Comment by NexonSU** [comment]
@LyuzhouZ unfortunately, this is not helps with ceiling mount setup.

**Comment by LyuzhouZ** [comment]
There may be environmental interference that causes the drifting. Try use NOLO in a different environment (another room, backyard, etc.) Play NoloTetris in Mobile VR mode https://www.nolovr.com/android-mobile-vr and see if the problem persists. You may also find more information on our FAQ page https://www.nolovr.com/faq

**Comment by NexonSU** [comment]
Nope, not even working with another PC and mobile device.
Controllers drifting around even if i don't touch them.

**Comment by LyuzhouZ** [comment]
@NexonSU Please shoot an email to support@nolovr.com and include a description of the issue you ran into. Our customer service will follow up on your case. :-)

**Comment by roceh** [comment]
Seeing the same thing, the position info is reliable but rotation needs to be constantly reset periodically due to drift.

**Comment by LyuzhouZ** [comment]
@roceh Try use NOLO in a different environment (another room, backyard, etc.) Play NoloTetris in Mobile VR mode https://www.nolovr.com/android-mobile-vr Check our FAQ page https://www.nolovr.com/faq If the problem persists, you may contact our customer service  support@nolovr.com
